packages/core/contracts/l2/polygon/OracleBaseTunnel.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```

Comment:
A few questions:
- Does polygon export this in a library of some kind that would allow us _not_ to copy it over?
- Do we need to intermingle our code with external code or should we separate it? Since this is becoming more common, could it make sense for us to have an "external" folder that includes code that we didn't write and/or aren't maintaining with READMEs for each section detailing where the code came from and a changelog of any modifications we made to fit our work.
- Does this code need to be audited? Or should we wall this off for the auditors?

Reply:
- no not yet
- yes we definitely should have an external code section
- the FX contracts are currently getting audited, so I don't know. Technically they are unaudited and would be safer for us to audit them. But its a lot of code.

Reply:
I like this idea of having an external section. this is the case in a bunch of places!